File: source/shop-backend/src/main/java/com/example/shop/domain/product/entity/Product.java
package com.example.shop.domain.product.entity;

import com.example.shop.domain.category.entity.Category;
import com.example.shop.domain.product.enums.ProductStatus;
import com.example.shop.global.common.BaseEntity;

import jakarta.persistence.*;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    private String name;
    private String description;
    private String brand;

    private int price;
    private Integer discountPrice;
    private int stock;

    @Enumerated(EnumType.STRING)
    private ProductStatus status;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ProductImage> images;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ProductOptionGroup> optionGroups;

    // Additional methods can be added here as needed
}

File: source/shop-admin/src/types/product.ts
export type ProductStatus = 'ACTIVE' | 'INACTIVE' | 'DELETED';

export interface ProductImage {
    imageId: number;
    imageUrl: string;
    sortOrder: number;
}

export interface Product {
    productId: number;
    name: string;
    description?: string;
    brand?: string;
    price: number;
    discountPrice?: number;
    stock: number;
    status: ProductStatus;
    category?: {
        categoryId: number;
        name: string;
    };
    images?: ProductImage[];
    createdAt?: string;
    updatedAt?: string;
}
